### PR TITLE
Set proper ingress defaults for galley meshcfg

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/17148 and adds tests to prevent regressions

### CODE REVIEW

File: galley/pkg/meshconfig/cache.go [hunk 1]
```diff
@@ -45,6 +45,13 @@ type FsCache struct {
 var _ Cache = &FsCache{}
 var _ io.Closer = &FsCache{}
 
+var DefaultMeshConfig = func() v1alpha1.MeshConfig {
```
Comment (hzxuzhonghu) [inline galley/pkg/meshconfig/cache.go]: duplicate with that in **`galley/pkg/config/meshcfg/defaults.go`**

**Comment by hzxuzhonghu** [comment]
BTW, is there a plan to deprecate ingress?

File: galley/pkg/meshconfig/cache.go (hunk 1 above)
Comment (howardjohn) [reply galley/pkg/meshconfig/cache.go]: @ozevren made that function not set these ingress settings. Not sure of the context, I think he would be able to answer intended behavior best? ref https://github.com/istio/istio/pull/17142

Or we could just change the default for everything not special case

Comment (ozevren) [reply galley/pkg/meshconfig/cache.go]: pkg/config/... is the new pipeline that I am going to be introducing enabling soon. This change needs to be reflected there as well.

**Comment by howardjohn** [comment]
@rshriram mind taking a look?

**Pull request merged**